api/handler/v1/project.go

```diff
@@ -2,25 +2,157 @@ package v1
 
 import (
 	"context"
+	"errors"
+	"strings"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
+	grpczap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/odpf/shield/internal/project"
 
 	shieldv1 "go.buf.build/odpf/gw/odpf/proton/odpf/shield/v1"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+var grpcProjectNotFoundErr = status.Errorf(codes.NotFound, "project doesn't exist")
+
+type ProjectService interface {
+	GetProject(ctx context.Context, id string) (project.Project, error)
+	CreateProject(ctx context.Context, project project.Project) (project.Project, error)
+	ListProject(ctx context.Context) ([]project.Project, error)
+	UpdateProject(ctx context.Context, toUpdate project.Project) (project.Project, error)
+}
+
 func (v Dep) ListProjects(ctx context.Context, request *shieldv1.ListProjectsRequest) (*shieldv1.ListProjectsResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+	logger := grpczap.Extract(ctx)
+	var projects []*shieldv1.Project
+
+	projectList, err := v.ProjectService.ListProject(ctx)
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcInternalServerError
+	}
+
+	for _, v := range projectList {
+		projectPB, err := transformProjectToPB(v)
+		if err != nil {
+			logger.Error(err.Error())
+			return nil, grpcInternalServerError
+		}
+
+		projects = append(projects, &projectPB)
+	}
+
+	return &shieldv1.ListProjectsResponse{Projects: projects}, nil
 }
 
 func (v Dep) CreateProject(ctx context.Context, request *shieldv1.CreateProjectRequest) (*shieldv1.CreateProjectResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+	logger := grpczap.Extract(ctx)
+	metaDataMap, err := mapOfStringValues(request.GetBody().Metadata.AsMap())
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcBadBodyError
+	}
+
+	slug := request.GetBody().Slug
```

Comment:
Would need validations as well that slug can't contain spaces or special chars except for hyphens and underscore. Also, let's do a lowercase() on it.